src/main/java/io/netty/incubator/channel/uring/SockaddrIn.java

```diff
@@ -115,33 +115,30 @@ static InetSocketAddress readIPv4(long memory, byte[] tmpArray) {
         }
     }
 
-    static InetSocketAddress readIPv6(long memory, byte[] tmpArray) {
-        assert tmpArray.length == 16;
-        int port = handleNetworkOrder(PlatformDependent.getShort(
-                memory + Native.SOCKADDR_IN6_OFFSETOF_SIN6_PORT)) & 0xFFFF;
-        PlatformDependent.copyMemory(
-                memory + Native.SOCKADDR_IN6_OFFSETOF_SIN6_ADDR + Native.IN6_ADDRESS_OFFSETOF_S6_ADDR,
-                tmpArray, 0, 16);
-        int scopeId = PlatformDependent.getInt(memory + Native.SOCKADDR_IN6_OFFSETOF_SIN6_SCOPE_ID);
-        try {
-            return new InetSocketAddress(Inet6Address.getByAddress(null, tmpArray, scopeId), port);
-        } catch (UnknownHostException ignore) {
-            return null;
-        }
-    }
+    static InetSocketAddress readIPv6(long memory, byte[] ipv6Array, byte[] ipv4Array) {
+        assert ipv6Array.length == 16;
+        assert ipv4Array.length == 4;
```

Comment:
We don't have constants for these values? Mainly so we can talk about them symbolically, rather than having to know from context what 16, 12, and 4 means every time they show up.

Reply:
Fixed... PTAL again